### ISSUE TITLE
 Return exception info from  `with_dodal_module` and `with_plan_module`

### ISSUE THREAD

**Issue by dperl-dls** [DiamondLightSource/blueapi#726]
Recently importing our plans (mx-bluesky/hyperion) into the BlueApi context started failing with a new (2.10) version of pydantic, which when running `create_model` on the plan arguments fails when one is a `dataclasses.dataclass` containing devices, because the devices don't implement `__get_pydantic_core_schema__`. This would have been substantially easier to diagnose if we had the information that exceptions had occurred during the above functions available to us when running them

## Acceptance Criteria
these methods have return values that allow the caller to determine if an exception happened and some info about it

**Comment by callumforrester** [comment]
@DiamondJoseph is this related to #727?

**Comment by dperl-dls** [comment]
@callumforrester yes, related in the sense that we weren't getting errors from these above-mentioned functions when setting up our context
